src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.ts

```diff
@@ -48,13 +48,25 @@ export class RgwBucketService {
     return this.http.get(`${this.url}/${bucket}`);
   }
 
-  create(bucket: string, uid: string, zonegroup: string, placementTarget: string) {
+  create(
+    bucket: string,
+    uid: string,
+    zonegroup: string,
+    placementTarget: string,
+    lockEnabled: boolean,
+    lockMode: 'GOVERNANCE' | 'COMPLIANCE',
+    lockRetentionPeriodDays: string,
+    lockRetentionPeriodYears: string
+  ) {
     let params = new HttpParams();
     params = params.append('bucket', bucket);
     params = params.append('uid', uid);
     params = params.append('zonegroup', zonegroup);
     params = params.append('placement_target', placementTarget);
-
+    params = params.append('lock_enabled', lockEnabled ? 'true' : 'false');
```

Comment:
I was curious if we could just use an object to create the right `HttpParams` object. The constructor of `HttpParams` does provide that :)

That's why I suggest to improve the readability by using the constructor attribute `fromObject` here instead of the append statements.

```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.ts b/src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.ts
index 44dc1e941c..71907497d7 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.ts
@@ -54,20 +54,24 @@ export class RgwBucketService {
     zonegroup: string,
     placementTarget: string,
     lockEnabled: boolean,
-    lockMode: 'GOVERNANCE' | 'COMPLIANCE',
-    lockRetentionPeriodDays: string,
-    lockRetentionPeriodYears: string
+    lock_mode: 'GOVERNANCE' | 'COMPLIANCE',
+    lock_retention_period_days: string,
+    lock_retention_period_years: string
   ) {
-    let params = new HttpParams();
-    params = params.append('bucket', bucket);
-    params = params.append('uid', uid);
-    params = params.append('zonegroup', zonegroup);
-    params = params.append('placement_target', placementTarget);
-    params = params.append('lock_enabled', lockEnabled ? 'true' : 'false');
-    params = params.append('lock_mode', lockMode);
-    params = params.append('lock_retention_period_days', lockRetentionPeriodDays);
-    params = params.append('lock_retention_period_years', lockRetentionPeriodYears);
-    return this.http.post(this.url, null, { params: params });
+    return this.http.post(this.url, null, {
+      params: new HttpParams({
+        fromObject: {
+          bucket,
+          uid,
+          zonegroup,
+          placement_target: placementTarget,
+          lock_enabled: String(lockEnabled),
+          lock_mode,
+          lock_retention_period_days,
+          lock_retention_period_years
+        }
+      })
+    });
   }
 
   update(
```

Reply:
Thx